File: ruby/magnitude_test.rb
require './magnitude.rb'

def assert_equal(a, b)
  str = Magnitude.format(a)
  raise "Failure #{str} doesn't equal #{b}" if str != b
end

assert_equal 0, '0'
assert_equal 10, '10'
assert_equal 23456, '23K'
assert_equal 312344000, '312M'
assert_equal 1454000000, '1B'
assert_equal 12344000000000, '12T'

File: scala/magnitude/MagnitudeTest.scala
package magnitude

object MagnitudeApp {

  def assertEqual(a: Long, b: String): Unit = {
    val str = Magnitude.format(a)
    if (!str.equals(b))
      throw new RuntimeException("Failed %s not equal to %s".format(str, b))
  }
  
  def main(args: Array[String]): Unit = {
    assertEqual(0, "0")
    assertEqual(10, "10")
    assertEqual(23456, "23K")
    assertEqual(312344000, "312M")
    assertEqual(1454000000, "1B")
    assertEqual(12344000000000L, "12T")
  }
}